internal/configmodels/configmodels.go

```diff
@@ -0,0 +1,112 @@
+which define the behavior. We also provide helper structs ReceiverSettings, ExporterSettings,
+ProcessorSettings, which define the common settings and un-marshaling from config files.
+
+Specific Receivers/Exporters/Processors are expected to at the minimum implement the
+corresponding interface and if they have additional settings they must also extend
+the corresponding common settings struct (the easiest approach is to embed the common struct).
+*/
+
+// ConfigV2 defines the configuration V2 for the various elements of collector or agent.
+type ConfigV2 struct {
+	Receivers  Receivers
+	Exporters  Exporters
+	Processors Processors
+	Pipelines  Pipelines
+}
+
+// Receiver is the configuration of a receiver. Specific receivers must implement this
+// interface and will typically embed ReceiverSettings struct or a struct that extends it.
+type Receiver interface {
+}
+
+// Receivers is a map of names to Receivers
+type Receivers map[string]Receiver
+
+// Exporter is the configuration of an exporter. Specific exporters must implement this
+// interface and will typically embed ExporterSettings struct or a struct that extends it.
+type Exporter interface {
+}
+
+// Exporters is a map of names to Exporters
+type Exporters map[string]Exporter
+
+// Processor is the configuration of a processor. Specific processors must implement this
+// interface and will typically embed ProcessorSettings struct or a struct that extends it.
+type Processor interface {
+}
+
+// Processors is a map of names to Processors
+type Processors map[string]Processor
+
+// DataType is the data type that is supported for collection. We currently support
+// collecting metrics and traces, this can expand in the future (e.g. logs, events, etc).
+type DataType int
+
+// Currently supported data types. Add new data types here when new types are supported in the future.
+const (
+	// TracesDataType is the data type tag for traces
+	TracesDataType DataType = iota
+
+	// MetricsDataType is the data type tag for metrics
+	MetricsDataType
+)
+
+// Pipeline defines a single pipeline.
+type Pipeline struct {
+	Name       string   `mapstructure:"-"`
+	InputType  DataType `mapstructure:"-"`
+	Receivers  []string `mapstructure:"receivers"`
+	Processors []string `mapstructure:"processors"`
+	Exporters  []string `mapstructure:"exporters"`
+}
+
+// Pipelines is a map of names to Pipelines
+type Pipelines map[string]*Pipeline
+
+// Below are common setting structs for Receivers, Exporters and Processors.
+// These are helper structs which you can embed when implementing your specific
+// receiver/exporter/processor config storage.
+
+// ReceiverSettings defines common settings for a single-protocol receiver configuration.
+// Specific receivers can embed this struct and extend it with more fields if needed.
+type ReceiverSettings struct {
+	Enabled  bool   `mapstructure:"enabled"`
+	Endpoint string `mapstructure:"endpoint"`
```

Comment:
An issue for down the line: it will be easier for users if agent defaults to `127.0.0.1:<port>` when agent is used as sidecar (or any configuration that somehow shares network namespace), and to `:<port>` when agent/collector is used as daemonset or own its pod (or any configuration in which it is "remote" to the senders).

Reply:
I believe this means that we need to supply these changes in the default config as config file bundled with either agent or collector.